### PR TITLE
Warn if .lein-env is present on launch (dev-only)

### PR DESCRIPTION
**Pull request by camsaul**
From https://github.com/metabase/metabase/wiki/Migrating-from-Leiningen-to-tools.deps#custom-env-var-values:

>  Environ will attempt to read the `.lein-env` file (Git-ignored by default) if one is present and prefer its values to the JVM system properties you've specified this way. `lein-environ` plugin will have created this file previously if you've ran with `lein` before. Be sure to delete this file if present so it doesn't ignore the properties you specify.

This PR is an attempt to fix this little gotcha so it stops biting people in the ass.

1. Warn on startup if `.lein-env` is found in the project root directory
2. ~~Remove `.lein-env` from `.gitignore` so people will notice it and remove it.~~ (update: not doing this anymore since it could be risky, see https://github.com/metabase/metabase/pull/17477#issuecomment-899756042)

### CODE REVIEW

**Comment by camsaul** [comment]
Alternatively... maybe we can just set up a Git hook to delete this file for us? I don't know how to do that tho.

Also, maybe just deleting the file if we see it makes more sense than complaining about it... it's too late at that point tho since Environ will already have read things from it

**Comment by camsaul** [comment]
Now that I think about it adding this to `.gitignore` sounds kinda dangerous since someone could accidentally stage it and then commit a file with potentially sensitive credentials... going to revert the `.gitignore` change.

File: src/metabase/config.clj [hunk 1]
```diff
@@ -127,3 +128,18 @@
 (def ^Keyword mb-session-cookie-samesite
   "Value for session cookie's `SameSite` directive. Must be one of \"none\", \"lax\", or \"strict\" (case insensitive)."
   (mb-session-cookie-samesite*))
+
+;; In 0.40.0 we switched from Leiningen to deps.edn. This warning here to keep people from being bitten in the ass by
```
Comment (jeff303) [inline src/metabase/config.clj:132]: 0.41.0, right?

**Review by jeff303 — APPROVED** [review]

File: src/metabase/config.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/config.clj:132]: oops. I will fix

**Comment by camsaul** [comment]
CodeCov is reporting this as only partially covered because only one branch of the  `(if-not is-prod? ...)` is taken during tests (which is to be expected...) but since this doesn't actually run in prod that's ok

**Pull request merged**